velox/exec/Window.cpp

```diff
@@ -41,16 +41,42 @@ void initKeyInfo(
   }
 }
 
-void checkDefaultWindowFrame(const core::WindowNode::Function& windowFunction) {
-  VELOX_CHECK_EQ(
-      windowFunction.frame.type, core::WindowNode::WindowType::kRange);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.startType,
-      core::WindowNode::BoundType::kUnboundedPreceding);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.endType, core::WindowNode::BoundType::kCurrentRow);
-  VELOX_CHECK_EQ(windowFunction.frame.startValue, nullptr);
-  VELOX_CHECK_EQ(windowFunction.frame.endValue, nullptr);
+// Update frameBound with the first row of the partition i.e zero for UNBOUNDED
+// PRECEDING frame bound.
+void findRowUnboundedPreceding(
+    vector_size_t* frameBound,
+    vector_size_t numRows) {
+  std::memset(frameBound, 0, numRows * sizeof(vector_size_t));
+}
+
+// Update frameBound with the first row of the peer group when frame start is
+// CURRENT ROW in RANGE mode.
+void findCurrentRowFrameStart(
+    vector_size_t* frameBound,
+    vector_size_t partitionStartRow,
```

Comment:
Why are you passing partitionStartRow if it is not needed ? Same comment for the following methods.

Reply:
Apologies for missing this, removed the parameter.